receiver_prod.go

```diff
@@ -0,0 +1,21 @@
+//go:build !debug
+// +build !debug
+
+package amqp
+
+// muxMsg sends the current decoded message to the channel of incoming messages.
+// it returns false if a client-side close has been initiated.
+func (r *Receiver) muxMsg() bool {
+	select {
+	case r.messages <- r.msg:
+		// message received
+		// NOTE: writing to this should NEVER block.
+		// if it does, it means we have a flow control
+		// bug so our peer sent a message exceeding
+		// the link credit
+		return true
+	case <-r.l.close:
```

Comment:
Should we also be checking r.l.done?

Reply:
No, that was the original bug which sadly got lost in the delta.

To recap, `r.l.done` is closed on when `r.mux()` exits.  And since `muxReceive()` is called from the mux, `r.l.done` will never be closed.

Reply:
Do you mind adding in a comment for that? I made the mistake of analyzing it isolation so it'd be easy to lose that context looking in there. 

By that same token it should never be used anywhere where that is not true (unlikely, but it's a slight footgun).

Reply:
Good point.  I added comments for the `done` channels explaining this.